cli/azd/main.go

```diff
@@ -39,7 +40,11 @@ func main() {
 
 	log.SetFlags(log.LstdFlags | log.Lshortfile)
 
-	if !isDebugEnabled() {
+	if isDebugEnabled() {
+		azcorelog.SetListener(func(event azcorelog.Event, msg string) {
```

Comment:
From https://github.com/Azure/azure-dev/pull/1051#discussion_r1009929307:

@vhvb1989 

> I would prefer not to have Azure SDK logs added by default to the azd logs. IMHO, it should be an opt in configuration. Something like `azd config logs azuresdkclient ON`

The whole point of the `debug` flag is to debug issues when bad things happen.. Wouldn't adding additional knobs for users make it harder to instruct users to attach debug logs?

If the main problem is noise, we can definitely consider adding a policy to provide only outgoing request and incoming response info, but currently we don't have any capabilities to debug which is problematic.